app/models/measurement/gr_value_loader.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+class Measurement
+  class GrValueLoader
+    include ActiveModel::Model
+    include ActiveRecord::AttributeAssignment
+
+    attr_accessor :measurement, :levels, :ministry_id, :assignment_id,
+                  :mcc, :period, :source, :historical
+
+    def load_gr_value
+      return load_historic_gr_values if @historical
+      @levels.each do |level|
+        measurement_level_id = @measurement.send("#{level}_id")
+        filter_params = gr_request_params(level)
+        gr_resp = GlobalRegistryClient.client(:measurement_type).find(measurement_level_id, filter_params)
+        value = gr_resp['measurement_type']['measurements'].map { |m| m['value'].to_f }.sum
```

Comment:
Can this be simplified to just `gr_resp['measurement_type']['measurements'].sum { |m| m['value'].to_f }`? See http://apidock.com/rails/Enumerable/sum